gax-grpc/src/main/java/com/google/api/gax/grpc/InstantiatingGrpcChannelProvider.java

```diff
@@ -113,12 +113,25 @@ private InstantiatingGrpcChannelProvider(Builder builder) {
 
   @Override
   public boolean needsExecutor() {
-    return executorProvider == null;
+    // Use the default gRPC executor
+    return false;
```

Comment:
This looks strange. It seems like it can break all existing manual overrides of the executors people may have already introduced. I.e. this method is called in `ClientContext.create()` method to figure out if a "custom" executor from an executor should be used (passed as executorProvider in stub settings). 

I guess in most cases it used to return `true`, which lead to using the executorProvider from stubSettings, which in its turn (unless user overrides it) leads to using the result of `InstantiatingExecutorProvider.newBuilder().build();` 

I guess if we are talking about default behavior before this change and after this change it is ok (the default behavior will change, but the new default behavior is supposed to be better).

But if somebody really knew what they where doing and decided to override executorProvider in their stubSettings it seems like it will be ignored now, because `transportChannelProvider.needsExecutor()` in `ClientContext.create()` will always return `false`, ignoring the custom executor.




But it is if nobody has overridden the `executorProvider` in the `stubSettings`. If somebody has done it

Reply:
This is the heart of the PR: the gax executor should not be shared with grpc. The PR proposes that channel executor should be configured on the channel providers instead of piping it through.

If you prefer to pipe it through, then the only approach I can think of is:
Add:
StubSettings#setTimerExecutor(ExecutorProvider)
StubSettings#setChannelExecutor(ChannelExecurtorProvider)

Have StubSettings#setExecutorProvider() call both the methods.

Add TransportChannelProvider#acceptsExecutor

Update ClientContext#create to have some logic like:
if (transportChannelProvider.needsExecutor() || (transportChannelProvider.acceptsExecutor() && channelExecutorProvider != null) {
   transportChannelProvider.withExecutor(channelExecutorProvider.get())
}

To me, this introduces a lot of complexity that is very hard to reason about